### ISSUE TITLE
Search Endpoint Decoding Error

### ISSUE THREAD

**Issue by FreddyStraub** [chojnac/NotionSwift#56]
#### Description
The `search` endpoint in NotionSwift is encountering a decoding error when processing the response. The issue appears to be related to a missing `id` key in the `select.options` field of the `Category` property. Here’s the error:

> `decodingError(Swift.DecodingError.keyNotFound(CodingKeys(stringValue: "id", intValue: nil), Swift.DecodingError.Context(codingPath: [CodingKeys(stringValue: "results", intValue: nil), _CodingKey(stringValue: "Index 2", intValue: 2), CodingKeys(stringValue: "properties", intValue: nil), _CodingKey(stringValue: "Category", intValue: nil), CodingKeys(stringValue: "select", intValue: nil), CodingKeys(stringValue: "options", intValue: nil), _CodingKey(stringValue: "Index 0", intValue: 0)], debugDescription: "No value associated with key CodingKeys(stringValue: "id", intValue: nil) ("id").", underlyingError: nil)))
> `

#### Steps to Reproduce
1. Perform a `search` request using the library.
2. Observe the decoding error in the response handling.

**Comment by FreddyStraub** [comment]
I think i also found the part of the responses that produces the error. It looks like that there can be database properties options that have no id:

![image](https://github.com/user-attachments/assets/32b5529b-bd05-4c72-bec1-7ee542a60b4e)

**Comment by FreddyStraub** [comment]
Managed to fix it locally by making the SelectOption Identifier optional:

```swift
extension DatabasePropertyType {
    public struct SelectOption {
        public typealias Identifier = EntityIdentifier<SelectOption, UUIDv4>
        public let name: String
        public let id: Identifier?          <--- This needed to be optional to make it work
        public let color: String

        public init(
            name: String,
            id: Identifier = .init(),
            color: String = "default"
        ) {
            self.name = name
            self.id = id
            self.color = color
        }
    }
}
```

**Comment by chojnac** [comment]
Can you please provide more details about your data? Based on the documentation `id` property is not optional and it always should be set. Also in my tests I always got `id` returned from server.

**Comment by FreddyStraub** [comment]
Yes, also checked the doc again and there should be an ID. The databases the produces the error is my **Readwise** DB. Readwise is a tool connected to Notion that saves kindle highlights and stuff. But then I probably have to issue that to the Notion API team? - That there currently can be Databases and Items without an ID.

**Comment by FreddyStraub** [comment]
Wrote the Notion API Support Team. Keep you updated here 😃

**Comment by FreddyStraub** [comment]
I clarified the issuse with the Notion API Support team and as it turned out they had bug and didn't return the id. The deployed a fix and now the API is delivering ids as expected and document in the Notion API Docs. Will close that issue here and abonden my PR.
